File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/shrines/ShrineBlock.java
package generations.gg.generations.core.generationscore.common.world.level.block.shrines;

import dev.architectury.registry.registries.RegistrySupplier;
import generations.gg.generations.core.generationscore.common.world.level.block.generic.GenericRotatableModelBlock;
import generations.gg.generations.core.generationscore.common.world.level.block.entities.MutableBlockEntityType;
import generations.gg.generations.core.generationscore.common.world.level.block.entities.shrines.ShrineBlockEntity;
import net.minecraft.resources.ResourceLocation;

public class ShrineBlock<T extends ShrineBlockEntity> extends GenericRotatableModelBlock<T> {
    protected ShrineBlock(Properties materialIn, RegistrySupplier<MutableBlockEntityType<T>> blockEntityFunction, ResourceLocation model, int width, int height, int length) {
        super(materialIn, blockEntityFunction, model, width, height, length);
        reassignStateDefinition();
        this.registerDefaultState(createDefaultState());
    }

    protected ShrineBlock(Properties materialIn, RegistrySupplier<MutableBlockEntityType<T>> blockEntityFunction, ResourceLocation model) {
        super(materialIn, blockEntityFunction, model);
    }

    public boolean isActivatable() {
        return false;
    }
}

File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/shrines/ShrineBlock.kt
package generations.gg.generations.core.generationscore.common.world.level.block.shrines

import generations.gg.generations.core.generationscore.common.world.level.block.entities.shrines.ShrineBlockEntity
import generations.gg.generations.core.generationscore.common.world.level.block.generic.GenericRotatableModelBlock
import net.minecraft.resources.ResourceLocation

abstract class ShrineBlock<T : ShrineBlockEntity>(
    properties: Properties,
    model: ResourceLocation? = null,
    width: Int = 0,
    height: Int = 0,
    length: Int = 0
) : GenericRotatableModelBlock<T>(properties = properties, model = model, width = width, height = height, length = length) {

    init {
        reassignStateDefinition()
        this.registerDefaultState(createDefaultState())
    }

    open val isActivatable: Boolean
        get() = false
}
